### PR TITLE
fix(require-explicit-slots): add support for type references

### LINKED ISSUE #2398: `require-explicit-slots` doesn't recognize interface passed to the macro
**Tell us about your environment**

- **ESLint version:** `8.56.0`
- **eslint-plugin-vue version:** `9.21.1`
- **Vue version:** `3.4.15`
- **Node version:** `20.10.0`
- **Operating System:** `Mac OS 14.2.1`

**Please show your full configuration:**

```json5
module.exports = {
  parser: 'vue-eslint-parser',

  parserOptions: {
    parser: '@typescript-eslint/parser',
  },

  extends: ['plugin:vue/base'],

  rules: {
    'vue/require-explicit-slots': 'error',
  },
};
```

**What did you do?**	

```vue
<template>
  <div>
    <slot />
  </div>
</template>

<script lang="ts" setup>
interface Slots {
  default: () => string;
}

defineSlots<Slots>();
</script>

```

**What did you expect to happen?**
No error reported.

**What actually happened?**
```
test.vue
  3:5  error  Slots must be explicitly defined  vue/require-explicit-slots

✖ 1 problem (1 error, 0 warnings)
```

**Repository to reproduce this issue**
https://stackblitz.com/edit/stackblitz-starters-reg1dy?file=test.vue

```
~$ npm run lint
```

### PR DESCRIPTION
**Pull request by chadwickellis**
This pull request resolves #2398 using an approach requested within #2484.

### CODE REVIEW

File: typings/eslint-plugin-vue/util-types/utils.ts [hunk 1]
```diff
@@ -191,6 +192,66 @@ export type ComponentEmit =
   | ComponentInferTypeEmit
   | ComponentUnknownEmit
 
+type ComponentArraySlotDetectName = {
```
Comment (ota-meshi) [inline typings/eslint-plugin-vue/util-types/utils.ts:195]: Hmm🤔. I think defineSlots only accepts Type literals or Interfaces or references that define function properties, so I don't think `type: 'array'` and `type: 'object'` are used at all. Am I wrong?
If there is logic or types that we don't use, can you remove them?

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR!
It looks almost good, but I have one comment.

File: typings/eslint-plugin-vue/util-types/utils.ts (hunk 1 above)
Comment (chadwickellis) [reply typings/eslint-plugin-vue/util-types/utils.ts]: Thank you for your feedback. I've removed the redundant type definitions and the associated utility method. Let me know if there's anything else that needs adjustment.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**